### ISSUE TITLE
ACKTR inconsistencies with paper

### ISSUE THREAD

**Issue by IvanStreltsov** [openai/baselines#130]
I noticed that there are some inconsistencies between ACKTR code and their original publication.

In the text they say that they use factored damping but in the code the variable is set to False.

Also I don't understand why do they compare ACKTR and A2C using different architectures. When I run A2C using baselines code, I get almost as good results as ACKTR and much better results than the ones that they report in their paper. From my perspective, there are no reasons to use different architectures in this case. When you report A2C results in your blog which version of A2C do you use? It is true that well tuned A2C can perform just as well as ACKTR?

Thank you very much!

**Comment by joschu** [comment]
@tonywu95

**Comment by tonywu95** [comment]
Hi @IvanStreltsov . 
Re: Damping. We used regular damping not factored damping in the code. We will fix it in the paper.
Re: Architecture. For A2C it's the regular DQN architecture, 32-64-64-512-x. For ACKTR, we changed the last conv layer to 32 for speeding up matrix inversion, 32-64-32-512-x.
Re: A2C performance. I'd be very surprised to see if A2C can as good as ACKTR, which also implies algorithms like PPO or ACER do not help either. The results we put in the paper are standard results trained using baselines code; it's shared across multiple publications (PPO, Equivalence Between Policy Gradients and Soft Q-Learning). I'd be very curious to know if you have some better hyperparameters.

**Comment by emansim** [comment]
I also quickly reran experiments with A2C and ACKTR both with 32 filters in last conv layer (cnn32) and 64 filters in last conv layer (cnn64) from current commit of openai baselines with 2 random seeds on BeamRider and Qbert and still getting the same results as in the paper.

![beamrider](https://user-images.githubusercontent.com/1727860/30357851-f3b1cb5e-980e-11e7-968e-866b413fc421.jpeg)

![qbert](https://user-images.githubusercontent.com/1727860/30357856-f6aa2ee6-980e-11e7-9c09-eb24d907ec07.jpeg)

**Comment by IvanStreltsov** [comment]
@emansim Did you use the last revision of A2C? I just cannot believe that I got so lucky with better random seeds.

**Comment by emansim** [comment]
@IvanStreltsov 

I also reran Breakout experiments and getting the same results as in the paper
![breakout](https://user-images.githubusercontent.com/1727860/30396440-a22241fe-9897-11e7-8e13-590ad4287367.jpeg)

Yes I was using the last revision of A2C. You can take a look at the code yourself https://github.com/emansim/baselines-mansimov
The only changes I made were dynamically assigning OPENAI_LOGDIR variable and changing num_timesteps variable from

```
num_timesteps = int(num_frames / 4 * 1.1) 
```

```
num_timesteps = int(num_frames // 4)
```

which I believe is a minor change.

For plotting please take a look at the following script https://github.com/emansim/baselines-mansimov/blob/master/baselines/a2c/visualize_atari.py and its dependency file https://github.com/emansim/baselines-mansimov/blob/master/baselines/a2c/load.py

**Comment by ahsteven** [comment]
@emansim Do you have a new link to the code for the plots? I am not able to reproduce the benchmark results.

**Comment by emansim** [comment]
@ahsteven 
https://github.com/ikostrikov/pytorch-a2c-ppo-acktr/blob/master/visualize.py

**Comment by rrmenon10** [comment]
@emansim The link provided doesn't provide the method to plot the confidence interval. Do you have a link to that?

**Comment by emansim** [comment]
@rrmenon10 use the following code for plotting https://github.com/ikostrikov/pytorch-a2c-ppo-acktr/blob/master/visualize.py
